File: C++/greatest_sub_range_1.cpp
#include <iostream>

using namespace std;

int main()
{
    int n;
	cin >> n;
	int a[n];
	for (int i = 0; i < n; i++)
		cin >> a[i];

	long long int ans = -10000000000000;

	for (int r = 1; r <= n; r++)
		for (int l = 0; l < r; l++)
		{
			long long int sum = 0;

			for (int i = l; i < r; i++)
				sum += a[i];

			if (sum > ans)
				ans = sum;
		}

	cout << ans << endl;
    return 0;
}

File: Python/greatest_sub_range_1.py
###first approach
n = int(input())
a = list(map(int, input().split(" ")))

first_try = True
for r in range(0, len(a) + 1):
    for l in range(r):
        if first_try == True:
            result = sum(a[l: r])
            first_try = False
        else:
            result = max(sum(a[l: r]), result)

print(result)

###second approach
n = int(input())
a = [int(x) for x in input().split()]
ans = -10**9
###-------you could use below code for ans instead of specify it hard code---------###
# import sys
# ans = -(sys.maxsize-1)
###-----or you could use below code------###
# ans = float('-inf')

for r in range(1, n + 1):
    for l in range(r):
        s = 0
        for i in range(l, r):
            s += a[i]
        if ans < s:
            ans = s

print(ans)